### PR TITLE
TEST-#5540: add module level setup function for ASV benchmarks

### LINKED ISSUE #5540: TEST: add module level `setup` function for all ASV benchmarks
`setup` function sets up the random number generator. Its call before each benchmark will improve the reproducibility of the results, since the generated data will no longer depend on the order in which the benchmark is called.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5540 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/benchmarks/hdk/benchmarks.py [hunk 1]
```diff
@@ -492,3 +492,6 @@ def time_groupby_agg_mean_dict(self, *args, **kwargs):
                 {col: "mean" for col in self.non_groupby_columns}
             )
         )
+
+
+from ..utils import setup  # noqa: E402, F401
```
Comment (Garra1980) [inline asv_bench/benchmarks/hdk/benchmarks.py:497]: I'd add a comment why we need this

Comment (anmyachev) [reply asv_bench/benchmarks/hdk/benchmarks.py:496]: Do you think that a comment inside `setup` function is not enough?

Comment (Garra1980) [reply asv_bench/benchmarks/hdk/benchmarks.py:496]: yeah, it is fine, i think my main concern is that we would need to not forget to add this one in each new benchmarks.py (say, for unidist backend) which is a bit inconvenient - maybe make it default?

Comment (anmyachev) [reply asv_bench/benchmarks/hdk/benchmarks.py:496]: I understand your concern, but however I don't know how to do it. Moreover, pandas also imports this function in every file. For now, this needs to be monitored.

**Comment by anmyachev** [comment]
@dchigarev could you take a look?

File: asv_bench/benchmarks/hdk/benchmarks.py [hunk 2]
```diff
@@ -492,3 +491,6 @@ def time_groupby_agg_mean_dict(self, *args, **kwargs):
...
```
Comment (dchigarev) [inline asv_bench/benchmarks/hdk/benchmarks.py:496]: why we import this at the end of the file?

Comment (anmyachev) [reply asv_bench/benchmarks/hdk/benchmarks.py:496]: I suppose not to accidentally override this function. A fairly common practice, for example, this is done in pandas.

Comment (anmyachev) [reply asv_bench/benchmarks/hdk/benchmarks.py:496]: Example: https://github.com/pandas-dev/pandas/blob/4965c51ef63dd06bcf460165a10bffa688142c46/asv_bench/benchmarks/algorithms.py#L192

@dchigarev any objections?

**Review by dchigarev — APPROVED** [review]

**Pull request merged**